dutree/dutree.py

```diff
@@ -389,6 +401,8 @@ def _scan_inner(self, files, children, fraction, a_or_u):
                 #   [Errno 13] Permission denied: '/run/user/1000/gvfs'
                 warnings.warn(str(e), OsWarning)
                 continue
+            exclude_xdev = self.xdev and not st.st_dev == self.starting_file_system 
+            exclude_skip_proc_sys = self.skip_proc_sys and st.st_dev in [self.proc_device, self.sys_device]
```

Comment:
Makes sense to me to make a device_blacklist and device_whitelist.

The scanner here won't have to consider *why* they're whitelisted.
```python
if xdev:
  blacklist = None
  whitelist = [starting_filesystem_dev_id]
elif skip_proc_sys:
  blacklist = [proc_dev_id, sys_dev_id]
  whitelist = None
else:
  blacklist = whitelist = None
```
^- pseudocode

And then here we can do:
```python
if whitelist and st.st_dev not in whitelist:
  pass
elif blacklist and st.st_dev in blacklist:
  pass
elif S_ISREG(st.st_mode):
  # do regular file stuff
elif S_ISDIR(st.st_mode):
  # do regular dir stuff
else:
  # the regular else
```